### ISSUE TITLE
[Bug] Terminating app due to uncaught exception 'SIGABRT', reason: 'Failed to marshal the Objective-C object 0x11c7151b0 (type: Xamarin_Forms_Platform_iOS_NavigationRenderer_Container). Could not find an existing managed instance for this object, nor was it possible to create a new managed instance (because the type 'Xamarin.Forms.Platform.iOS.NavigationRenderer+Container' does not have a constructor that takes one IntPtr argument).'

### ISSUE THREAD

**Issue by JulesMoorhouse** [xamarin/Xamarin.Forms#12059]
### Description
We're getting several of crash reports on iOS across several devices. We couldn't reproduce and trace the error, here is the error log from App Center

Hardware Model:      iPhone11,8
Version:         1.7.4 (1.7.4)
Code Type:       arm64e
Parent Process:  ??? [1]
OS Version:      iPhone OS 13.6 (17G68)
Report Version:  104

Exception Type:  SIGABRT
Exception Codes: #0 at 0x18135cdf0
Crashed Thread:  0

Application Specific Information:
*** Terminating app due to uncaught exception 'SIGABRT', reason: 'Failed to marshal the Objective-C object 0x11c7151b0 (type: Xamarin_Forms_Platform_iOS_NavigationRenderer_Container). Could not find an existing managed instance for this object, nor was it possible to create a new managed instance (because the type 'Xamarin.Forms.Platform.iOS.NavigationRenderer+Container' does not have a constructor that takes one IntPtr argument).'

Xamarin Exception Stack:
ObjCRuntime.RuntimeException: Failed to marshal the Objective-C object 0x11c7151b0 (type: Xamarin_Forms_Platform_iOS_NavigationRenderer_Container). Could not find an existing managed instance for this object, nor was it possible to create a new managed instance (because the type 'Xamarin.Forms.Platform.iOS.NavigationRenderer+Container' does not have a constructor that takes one IntPtr argument).
  at ObjCRuntime.Runtime.MissingCtor (System.IntPtr ptr, System.IntPtr klass, System.Type type, ObjCRuntime.Runtime+MissingCtorResolution resolution) <0x105eec3e0 + 0x00154> in <75e8f2cafde249969958834208668188#b6ff2b42dd217351f1823f7118768cad>:0
  at ObjCRuntime.Runtime.ConstructNSObject[T] (System.IntPtr ptr, System.Type type, ObjCRuntime.Runtime+MissingCtorResolution missingCtorResolution) <0x105eec5e0 + 0x0006b> in <75e8f2cafde249969958834208668188#b6ff2b42dd217351f1823f7118768cad>:0
  at ObjCRuntime.Runtime.GetNSObject[T] (System.IntPtr ptr) <0x105eed140 + 0x00177> in <75e8f2cafde249969958834208668188#b6ff2b42dd217351f1823f7118768cad>:0
  at UIKit.UIView.get_Superview () <0x105f4c710 + 0x00083> in <75e8f2cafde249969958834208668188#b6ff2b42dd217351f1823f7118768cad>:0
  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].UpdateParentPageAccessibilityElements () <0x10613a970 + 0x0007f> in <441b7767f9c446ffaba8ecc9de302557#b6ff2b42dd217351f1823f7118768cad>:0
  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].OnElementPropertyChanged (System.Object sender, System.ComponentModel.PropertyChangedEventArgs e) <0x10613a0d0 + 0x0030f> in <441b7767f9c446ffaba8ecc9de302557#b6ff2b42dd217351f1823f7118768cad>:0
  at (wrapper delegate-invoke) <Module>.invoke_void_object_PropertyChangedEventArgs(object,System.ComponentModel.PropertyChangedEventArgs)
  at Xamarin.Forms.BindableObject.OnPropertyChanged (System.String propertyName) <0x105483aa0 + 0x00087> in <3015a2ac424d4bfc9c78e6a5d3a3fc66#b6ff2b42dd217351f1823f7118768cad>:0
  at Xamarin.Forms.Element.OnPropertyChanged (System.String propertyName) <0x1054c5a30 + 0x0002f> in <3015a2ac424d4bfc9c78e6a5d3a3fc66#b6ff2b42dd217351f1823f7118768cad>:0
  at Xamarin.Forms.BindableObject.SetValueActual (Xamarin.Forms.BindableProperty property, Xamarin.Forms.BindableObject+BindablePropertyContext context, System.Object value, System.Boolean currentlyApplying, Xamarin.Forms.Internals.SetValueFlags attributes, System.Boolean silent) <0x105484c10 + 0x002df> in <3015a2ac424d4bfc9c78e6a5d3a3fc66#b6ff2b42dd217351f1823f7118768cad>:0
  at Xamarin.Forms.BindableObject.SetValueCore (Xamarin.Forms.BindableProperty property, System.Object value, Xamarin.Forms.Internals.SetValueFlags attributes, Xamarin.Forms.BindableObject+SetValuePrivateFlags privateAttributes) <0x105484620 + 0x004ef> in <3015a2ac424d4bfc9c78e6a5d3a3fc66#b6ff2b42dd217351f1823f7118768cad>:0
  at Xamarin.Forms.BindingExpression.ApplyCore (System.Object sourceObject, Xamarin.Forms.BindableObject target, Xamarin.Forms.BindableProperty property, System.Boolean fromTarget) <0x10548e320 + 0x00633> in <3015a2ac424d4bfc9c78e6a5d3a3fc66#b6ff2b42dd217351f1823f7118768cad>:0
  at Xamarin.Forms.BindingExpression.Apply (System.Boolean fromTarget) <0x10548dfb0 + 0x0008f> in <3015a2ac424d4bfc9c78e6a5d3a3fc66#b6ff2b42dd217351f1823f7118768cad>:0
  at Xamarin.Forms.BindingExpression+BindingExpressionPart.<PropertyChanged>b__49_0 () <0x105492f00 + 0x00023> in <3015a2ac424d4bfc9c78e6a5d3a3fc66#b6ff2b42dd217351f1823f7118768cad>:0
  at Xamarin.Forms.BindingExpression+BindingExpressionPart.PropertyChanged (System.Object sender, System.ComponentModel.PropertyChangedEventArgs args) <0x105492910 + 0x00293> in <3015a2ac424d4bfc9c78e6a5d3a3fc66#b6ff2b42dd217351f1823f7118768cad>:0
  at Xamarin.Forms.BindingExpression+WeakPropertyChangedProxy.OnPropertyChanged (System.Object sender, System.ComponentModel.PropertyChangedEventArgs e) <0x105492050 + 0x0005b> in <3015a2ac424d4bfc9c78e6a5d3a3fc66#b6ff2b42dd217351f1823f7118768cad>:0
  at (wrapper delegate-invoke) <Module>.invoke_void_object_PropertyChangedEventArgs(object,System.ComponentModel.PropertyChangedEventArgs)
  at GalaSoft.MvvmLight.ObservableObject.RaisePropertyChanged (System.String propertyName) <0x1067f6840 + 0x00083> in <4bfe803b47234f56b6c30e8bc3e0a4b9#b6ff2b42dd217351f1823f7118768cad>:0
  at myapp.ViewModel.BookingDetailsViewModel.set_EditBookingType (myapp.Enums.EditBookingType value) <0x1062d4e70 + 0x00067> in <08feac2aa17f40f7a6ee2a286d750a60#b6ff2b42dd217351f1823f7118768cad>:0
  at myapp.ViewModel.BookingDetailsViewModel.AddEditToolbarItem () <0x1062d5400 + 0x006b3> in <08feac2aa17f40f7a6ee2a286d750a60#b6ff2b42dd217351f1823f7118768cad>:0
  at myapp.Pages.BookingDetailsPage.OnAppearing () <0x10638c000 + 0x00193> in <08feac2aa17f40f7a6ee2a286d750a60#b6ff2b42dd217351f1823f7118768cad>:0
  at Xamarin.Forms.Page.SendAppearing () <0x105527a90 + 0x0018b> in <3015a2ac424d4bfc9c78e6a5d3a3fc66#b6ff2b42dd217351f1823f7118768cad>:0
  at Xamarin.Forms.Platform.iOS.PageRenderer.ViewDidAppear (System.Boolean animated) <0x106173d40 + 0x00103> in <441b7767f9c446ffaba8ecc9de302557#b6ff2b42dd217351f1823f7118768cad>:0
  at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) <0x105f28080 + 0x0002f> in <75e8f2cafde249969958834208668188#b6ff2b42dd217351f1823f7118768cad>:0
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) <0x105f27ff0 + 0x00043> in <75e8f2cafde249969958834208668188#b6ff2b42dd217351f1823f7118768cad>:0
  at myapp.iOS.Application.Main (System.String[] args) <0x105038540 + 0x00023> in <f3baac83ae5c4bd88aadac20ade85cdd#b6ff2b42dd217351f1823f7118768cad>:0

Were using Xamarin Forms version 4.5.0.396

Any help appreciated.

Thanks in advance.

**Comment by StephaneDelcroix** [comment]
Could you make sure your Xamarin.iOS version is up-to-date ? One way to workaround this bug might be to disable the linker for your project on Release builds

**Comment by JulesMoorhouse** [comment]
We already have the Linker set too `Don't Link` for iOS.

Also not sure how to tell what version of Xamarin.iOS was being used for the release version which is showing the crash.
Is it stored in a xml file that we could check? we will have that version of the code tagged in git.
Assuming it's useful to determine the Xamarin.iOS version ?

**Comment by JulesMoorhouse** [comment]
@StephaneDelcroix can you comment further please?

**Comment by PureWeen** [comment]
@JulesMoorhouse this looks like there's a property changed event firing against a page that's been disposed

```C#
.invoke_void_object_PropertyChangedEventArgs(object,System.ComponentModel.PropertyChangedEventArgs)
at GalaSoft.MvvmLight.ObservableObject.RaisePropertyChanged (System.String propertyName) <0x1067f6840 + 0x00083> in <4bfe803b47234f56b6c30e8bc3e0a4b9#b6ff2b42dd217351f1823f7118768cad>:0
at myapp.ViewModel.BookingDetailsViewModel.set_EditBookingType (myapp.Enums.EditBookingType value) <0x1062d4e70 + 0x00067> in <08feac2aa17f40f7a6ee2a286d750a60#b6ff2b42dd217351f1823f7118768cad>:0
at myapp.ViewModel.BookingDetailsViewModel.AddEditToolbarItem () <0x1062d5400 + 0x006b3> in<08feac2aa17f40f7a6ee2a286d750a60#b6ff2b42dd217351f1823f7118768cad>:0
at myapp.Pages.BookingDetailsPage.OnAppearing () <0x10638c000 + 0x00193> in
```

Indicates there's an action occurring during a page firing OnAppearing

It's hard to know much more without a repro or seeing your app

**Comment by Redth** [comment]
Since we haven't heard from you in more than 30 days, we hope this issue is no longer affecting you. If it is, please reopen this issue and provide the requested information so that we can look into it further. Thank you!
